### PR TITLE
Fix for Locations Filter

### LINKED ISSUE #2445: Open Y Location Filters Broken
"Step 2" of the built in Membership Calculator is populated by the following config page (/admin/config/content/location-filter-settings).  I believe this is meant to allow site owners to be able to include or exclude the branches / camps from the filtering system.

I'm working with a YMCA who has multiple branches that are not "full service membership" branches, and so, would only like to show a few of their branches in the membership calculator.  However, it seems you can only select All or None of the branches/camps.

Steps to recreate:

1) Visit [Sandbox](https://sandbox-carnation-cus.openy.org/admin/config/content/location-filter-settings) and Navigate to the Open Y Locations Filter Page (/admin/config/content/location-filter-settings).

2) Uncheck "All" and select any combination of the remaining options.  Click "Save".

3) The page will reload will all options checked.

Asking for advice or other ideas.  Thanks!!

### PR DESCRIPTION
**Pull request by podarok**
Original Issue, this PR is going to fix: #2445 

Make sure these boxes are checked before asking for review of your pull request - thank you!

If there is a new feature or this is a bug fix - use 9.x-2.x branch. We'll tag for release if the bug is critical asap or tag for release next bug fix release until critical issue arrived.

## Steps for review

- [ ] Please provide steps for review here.
- [ ] Please provide steps for review here.

## General checks
- [ ] All coding styles are fulfilled and there are no any issues reported by CodeSniffer. See [Code of Conduct](https://github.com/ymcatwincities/openy/wiki/Open-Y-Code-of-Conduct-and-Best-Practices).
- [ ] [Documentation](https://github.com/ymcatwincities/openy/tree/9.x-2.x/docs) has been updated according to PR changes.
- [ ] [Steps for review](https://github.com/ymcatwincities/openy/pull/94#issue-204580200) have been provided according to PR changes. <br/>Steps for review
- [ ] Make sure you've provided all necessary hook\_update\_N to [support upgrade path](https://github.com/ymcatwincities/openy/blob/9.x-2.x/docs/Development/Upgrade%20path.md).
- [ ] Make sure your git email is associated with account on drupal.org, otherwise you won't get commits there. <br/>drupal.org email
- [ ] If you would like to get credits on drupal.org, [check documentation](https://github.com/ymcatwincities/openy/blob/9.x-2.x/docs/Development/Contributing.md#drupalorg-credits).

Thank you for your contribution!

### CODE REVIEW

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

Check Open Y Installation Wizard at http://install.openy.cibox.tools/build3170/install.php

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/3170/

**Comment by bjwilkins** [comment]
Open Y Upgrade Path site check installed at http://upgrade.openy.cibox.tools/build3170/

**Comment by podarok** [comment]
Tested on build
image

File: modules/openy_features/openy_location/modules/openy_loc_filter/src/Form/LocationFilterSettingsForm.php [hunk 1]
```diff
@@ -126,11 +126,17 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
     /** @var \Drupal\Core\Config\Config $config */
     $config = $this->config(self::CONFIG_NAME);
     $locations = $form_state->getValue('locations');
-    if (isset($locations['All'])) {
-      $branches_list = $this->getBranchesList();
-      $locations = $branches_list['branch'] + $branches_list['camp'];
+
+    $branches_list = $this->getBranchesList();
```
Comment (hamrant) [inline modules/openy_features/openy_location/modules/openy_loc_filter/src/Form/LocationFilterSettingsForm.php]: Looks like `$branches_list` variable not used in the function

**Review by hamrant — CHANGES_REQUESTED** [review]

**Comment by froboy** [comment]
Thanks @podarok

**Review by froboy — APPROVED** [review]
Apart from @hamrant's comment this looks good.

File: modules/openy_features/openy_location/modules/openy_loc_filter/src/Form/LocationFilterSettingsForm.php (hunk 1 above)
Comment (podarok) [reply modules/openy_features/openy_location/modules/openy_loc_filter/src/Form/LocationFilterSettingsForm.php]: removed a line, nice catch

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

Check Open Y Installation Wizard at http://install.openy.cibox.tools/build3171/install.php

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/3171/

**Comment by bjwilkins** [comment]
Open Y Upgrade Path site check installed at http://upgrade.openy.cibox.tools/build3171/

**Review by hamrant — APPROVED** [review]

**Pull request merged**